crates/noirc_evaluator/src/brillig/brillig_gen/brillig_block.rs

```diff
@@ -697,11 +714,27 @@ impl<'block> BrilligBlock<'block> {
         let binary_type =
             type_of_binary_operation(dfg[binary.lhs].get_type(), dfg[binary.rhs].get_type());
 
-        let left = self.convert_ssa_register_value(binary.lhs, dfg);
-        let right = self.convert_ssa_register_value(binary.rhs, dfg);
+        let mut left = self.convert_ssa_register_value(binary.lhs, dfg);
+        let mut right = self.convert_ssa_register_value(binary.rhs, dfg);
 
         let brillig_binary_op =
-            convert_ssa_binary_op_to_brillig_binary_op(binary.operator, binary_type);
+            convert_ssa_binary_op_to_brillig_binary_op(binary.operator, &binary_type);
+
+        // Some binary operations with fields are issued by the compiler, such as loop comparisons, cast those to the bit size here
```

Comment:
The use of mut variables indicates this might have some helper method possible, but just something to keep in mind for future passes

Reply:
This should be removed when we stop generating Field comparisons for loops upstream, so I thought that this way it was easily removable

Reply:
This comment reads like the compiler is inserting instructions of the wrong type when creating loops. If that's the case we should fix the problem at the root rather than only fixing the brillig generated from it.

Reply:
Yes, this is related to https://github.com/noir-lang/noir/issues/1977 , where the compiler accepts as a valid type for the range of the loop a field type, so the inherent comparison between the iterator and the range is a comparison of fields. I was told when discussing this with @kevaundray that not allowing field ranges & iterators would be a rather big breaking change, so I think supporting it until fixed upstream with a breaking change could be the best option

Reply:
Opened bug to track https://github.com/noir-lang/noir/issues/1979

Reply:
Thanks for the issue. Let's come back to it at a different time then.